data-plane/receiver/src/test/java/dev/knative/eventingkafkabroker/receiver/integration/ReceiverVerticleTest.java

```diff
@@ -0,0 +1,260 @@
+package dev.knative.eventingkafkabroker.receiver.integration;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.FloatNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import dev.knative.eventingkafkabroker.receiver.CloudEventRequestToRecordMapper;
+import dev.knative.eventingkafkabroker.receiver.HttpVerticle;
+import dev.knative.eventingkafkabroker.receiver.ProducerDriver;
+import dev.knative.eventingkafkabroker.receiver.RequestHandler;
+import io.cloudevents.CloudEvent;
+import io.cloudevents.core.message.Message;
+import io.cloudevents.core.v1.CloudEventBuilder;
+import io.cloudevents.http.vertx.VertxHttpClientRequestMessageVisitor;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import io.vertx.kafka.client.producer.KafkaProducerRecord;
+import io.vertx.kafka.client.producer.impl.KafkaProducerRecordImpl;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(VertxExtension.class)
+public class ReceiverVerticleTest {
+
+  static final int TIMEOUT = 10;
+  static final int PORT = 8080;
+
+  @Test
+  public void requestHandler(final Vertx vertx, final VertxTestContext testContext)
+      throws Throwable {
+
+    final var httpClient = vertx.createHttpClient();
+    final var testCases = requestStream();
+    final var countDown = new CountDownLatch(testCases.size());
+    final var flags = testContext.checkpoint(testCases.size() * 2);
+
+    final var producerDriver = new ProducerDriver<String, CloudEvent>();
+    final var producer = producerDriver.producer(false);
+
+    final var handler = new RequestHandler<>(producer, new CloudEventRequestToRecordMapper());
+
+    final var httpServerOptions = new HttpServerOptions();
+    httpServerOptions.setPort(PORT);
+    final var verticle = new HttpVerticle(httpServerOptions, handler);
+
+    vertx.deployVerticle(verticle, testContext.succeeding(
+        h -> testCases.parallelStream().map(tc -> tc.requestResponse).forEach(requestResponse -> {
```

Comment:
How about a parametrized test with `@ParametrizedTest`?

```
@ParameterizedTest
@MethodSource("requestStream")
public void requestHandler(TestCase case, final Vertx vertx, final VertxTestContext testContext)
```

Two things:

* Avoid creating test cases objects, It's not really junit5 pragmatic... You can use `Stream<Arguments>`: https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests-sources-MethodSource
* IMPORTANT: Put the parametrized arguments before the Vertx and VertxTestContext arguments

Reply:
My goal was to send multiple messages, in this way we send only one at a time.

Reply:
you can always bootstrap the environment with setup and teardown methods, but this way i can run a particular test with particular arguments straight from the IDE

Reply:
https://junit.org/junit5/docs/current/user-guide/#writing-tests-classes-and-methods

Reply:
I solved extracting a helper method because junit annotations don't play nicely with vertx extension.